spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/dynamodb/DynamoDbAutoConfigurationTest.java

```diff
@@ -48,71 +51,153 @@
+						.isInstanceOf(CustomDynamoDBDynamoDbTableNameResolver.class);
 
-					assertThat(dynamoDbTableSchemaResolver).isNotNull();
-					assertThat(dynamoDBDynamoDbTableNameResolver).isNotNull();
+			});
+		}
+
+		@Test
+		void doesNotCreateDaxClientWhenDaxNotInClasspath() {
+			contextRunner.withClassLoader(new FilteredClassLoader(ClusterDaxClient.class)).run(context -> {
+				assertThat(context).hasSingleBean(DynamoDbClient.class);
+				assertThat(context.getBean(DynamoDbClient.class)).isNotInstanceOf(ClusterDaxClient.class);
+			});
+		}
 
-					assertThat(dynamoDbTableSchemaResolver)
-							.isInstanceOf(CustomDynamoDBDynamoDbTableSchemaResolver.class);
-					assertThat(dynamoDBDynamoDbTableNameResolver)
-							.isInstanceOf(CustomDynamoDBDynamoDbTableNameResolver.class);
+		@Test
+		void withDynamoDbClientCustomEndpoint() {
+			contextRunner.withPropertyValues("spring.cloud.aws.dynamodb.endpoint:http://localhost:8090")
+					.run(context -> {
+						assertThat(context).hasSingleBean(DynamoDbClient.class);
+						assertThat(context).hasSingleBean(DynamoDbTemplate.class);
+						assertThat(context).hasSingleBean(DynamoDbEnhancedClient.class);
 
-				});
+						ConfiguredAwsClient client = new ConfiguredAwsClient(context.getBean(DynamoDbClient.class));
+						assertThat(client.getEndpoint()).isEqualTo(URI.create("http://localhost:8090"));
+						assertThat(client.isEndpointOverridden()).isTrue();
+					});
+		}
+
+		@Test
+		void dynamoDbClientConfiguredSinceNoUrl() {
+			contextRunner.run(context -> {
+				assertThat(context).hasSingleBean(DynamoDbClient.class);
+				assertThat(context).hasSingleBean(DynamoDbTemplate.class);
+				assertThat(context).hasSingleBean(DynamoDbEnhancedClient.class);
+				assertThat(context).doesNotHaveBean(ClusterDaxClient.class);
+			});
+		}
+
+		@Test
+		void customDynamoDbClientConfigurer() {
+			contextRunner.withUserConfiguration(DynamoDbAutoConfigurationTest.CustomAwsClientConfig.class)
+					.run(context -> {
+						ConfiguredAwsClient dynamoDbClient = new ConfiguredAwsClient(
+								context.getBean(DynamoDbClient.class));
+						assertThat(dynamoDbClient.getApiCallTimeout()).isEqualTo(Duration.ofMillis(1999));
+						assertThat(dynamoDbClient.getSyncHttpClient()).isNotNull();
+					});
+		}
 	}
 
-	@Test
-	void customDynamoDbClientConfigurer() {
-		this.contextRunner.withUserConfiguration(DynamoDbAutoConfigurationTest.CustomAwsClientConfig.class)
-				.run(context -> {
-					ConfiguredAwsClient sesClient = new ConfiguredAwsClient(context.getBean(DynamoDbClient.class));
-					assertThat(sesClient.getApiCallTimeout()).isEqualTo(Duration.ofMillis(1999));
-					assertThat(sesClient.getSyncHttpClient()).isNotNull();
-				});
+	@Nested
+	class DaxTests {
+
+		@Test
+		void customClusterDaxClientDaxSettings() {
+			contextRunner.withPropertyValues(
+					"spring.cloud.aws.dynamodb.dax.url:dax://something.dax-clusters.us-east-1.amazonaws.com",
+					"spring.cloud.aws.dynamodb.dax.writeRetries:4",
```

Comment:
No camel case for properties. For example:

`spring.cloud.aws.dynamodb.dax.write-retries`